src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -279,6 +290,16 @@ const reducer = (state: InitialState, action: ActionTypes) => {
     case WalletActions.RESET_STATE:
       const resetProperties = omit(initialState, ['keyring', 'adapters', 'modal', 'deviceId'])
       return { ...state, ...resetProperties }
+    case WalletActions.RESET_LAST_DEVICE_INTERACTION_STATE: {
+      const { deviceState } = state
+      return {
+        ...state,
+        deviceState: {
+          ...deviceState,
+          lastDeviceInteractionStatus: undefined,
+        },
+      }
+    }
```

Comment:
Do we really need this?

Can't we just use `SET_DEVICE_STATE` with a payload of `{ lastDeviceInteractionStatus: undefined }`?

Reply:
You can't, because `SET_DEVICE_STATE` use the current state value as default value in case the state is not provided in the payload, it permits to avoid to add every states when you use it

If you do this: `setDeviceState({ lastDeviceInteractionStatus: undefined })`, it will keep the current value of `lastDeviceInteractionStatus` which could be `true`, `false`, `anything else`...

Reply:
Right you are, `SET_DEVICE_STATE` doesn't currently support explicitly setting `undefined` values, which is a design flaw.

This is not a concern for this PR, but something we should fix.

I added a TODO in https://github.com/shapeshift/web/pull/2589/commits/3e90f5ca2601ab71c136907bbcaa68664f141106